actix-api/src/actors/chat_server.rs

```diff
@@ -117,21 +102,40 @@ impl Handler<JoinRoom> for ChatServer {
     ) -> Self::Result {
         self.leave_rooms(&session);
 
-        if !self.rooms.contains_key(&room) {
-            self.rooms.insert(
-                room.clone(),
-                vec![session.clone()]
-                    .into_iter()
-                    .collect::<HashSet<String>>(),
-            );
-        }
+        let subject = format!("room.{}.*", room);
+        let queue = format!("{}-{}", session, room);
+        let session_recipient = self.sessions.get(&session).unwrap().clone();
+        let room_clone = room.clone();
+        let session_clone = session.clone();
+        let sub = self
+            .nc
+            .queue_subscribe(&subject, &queue)
+            .unwrap()
+            .with_handler(move |msg| {
+                if msg.subject == format!("room.{}.{}", room_clone, session_clone) {
+                    return Ok(());
+                }
+                let msg = MediaPacket::parse_from_bytes(&msg.data).unwrap();
+                let msg = Message {
+                    nickname: Arc::new(Some(msg.email.clone())),
+                    msg: Arc::new(msg),
+                };
+
+                match session_recipient.try_send(msg) {
```

Comment:
This looks like a .map_err instead of a match IMO, if there's an error Imo we should tear down the session altogether.